mapeditor/resourceExtractor/ResourceConverter.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * ResourceConverter.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "StdInc.h"
+
+#include "ResourceConverter.h"
+
+#include "../lib/JsonNode.h"
+#include "../lib/VCMIDirs.h"
+#include "../lib/filesystem/Filesystem.h"
+
+#include "BitmapHandler.h"
+#include "Animation.h"
+
+#include "boost/filesystem/path.hpp"
+#include "boost/locale.hpp"
+
+void ResourceConverter::convertExtractedResourceFiles(ConversionOptions conversionOptions)
+{
+	if (conversionOptions.splitDefs)
+		splitDefFiles(conversionOptions.deleteOriginals);
+
+	if (conversionOptions.convertPcxToPng)
+		doConvertPcxToPng(conversionOptions.deleteOriginals);
+}
+
+void ResourceConverter::doConvertPcxToPng(bool deleteOriginals)
+{
+	std::string filename;
+
+	bfs::path imagesPath = VCMIDirs::get().userExtractedPath() / "IMAGES";
+	bfs::directory_iterator end_iter;
+
+	for(bfs::directory_iterator dir_itr(imagesPath); dir_itr != end_iter; ++dir_itr)
+	{
+		try
+		{
+			if (!bfs::is_regular_file(dir_itr->status()))
+				return;
```

Comment:
@krs0 is it correct to use `return` instead of `continue` here?

Reply:
Yup, its a mistake. Works anyway because we have only images, but makes no sense. Will change to continue